tangelo/algorithms/projective/quantum_imaginary_time.py

```diff
@@ -310,6 +310,7 @@ def get_resources(self):
         resources["qubit_hamiltonian_terms"] = len(self.qubit_hamiltonian.terms)
         resources["pool_size"] = len(self.pool_operators)
         resources["circuit_width"] = self.final_circuit.width
+        resources["circuit_depth"] = self.final_circuit.depth()
         resources["circuit_gates"] = self.final_circuit.size
         resources["circuit_2qubit_gates"] = self.final_circuit.counts.get("CNOT", 0)
```

Comment:
Since you are touching this function anyway. It has been on our todos to make the 2qubit gates more comprehensive. You could use https://github.com/goodchemistryco/Tangelo/blob/c88a97ef4e16a5e8d7d5c718eb7ae00d195c166d/tangelo/linq/gate.py#L25.

Although many of these could be in theory more than 2 qubit gates. This may be better done as a method in `Circuit`. And probably even better in Circuit.add_gate(). Something like adding the line
```python3
def Circuit:
     def __init__(self, gates: List[Gate] = None, n_qubits=None, name="no_name"):
          ...
          self._n_qubit_gate_counts = dict()
      ...
     def add_gate(self, g):
          ...
          n_qubit = len(g.control) + len(g.target)
          self._n_qubit_gate_counts[n_qubit] = self._n_qubit_gate_counts.get(n_qubit, 0) + 1
```

Reply:
Counts for individual gates are already tracked in `self._gate_counts`.

I think I like James's idea: since Tangelo allows the user to build a circuit with any kind of gate (it's not a problem until it's converted to a particular format which may or may not support these gates), maybe it's better not to restrict ourselves to a list of predefined ones. Counting the number of qubits involved and then incrementing the corresponding count seems good and cheap.

Reply:
Implemented as above for all n qubit gates, `get_resources` displays only the 2 qubit gate count